static/js/Autocomplete.jsx

```diff
@@ -0,0 +1,489 @@
+    }, []);
+   const { onBlur, onKeyDown, ...otherDownShiftProps } = getInputProps();
+
+    const handleSearchKeyDown = (event) => {
+      onKeyDown(event)
+      if (event.keyCode !== 13) return;
+      const highlightedItem = highlightedIndex > -1 ? suggestions[highlightedIndex] : null
+      if (highlightedItem  && highlightedItem.type != 'search'){
+        redirectToObject(highlightedItem);
+        return;
+      }
+      const inputQuery = otherDownShiftProps.value
+      if (!inputQuery) return;
+      submitSearch(inputQuery);
+    };
+
+
+    const handleSearchButtonClick = (event) => {
+      const inputQuery = otherDownShiftProps.value
+      if (inputQuery) {
+        submitSearch(inputQuery);
+      } else {
+        focusSearch()
+      }
+    };
+
+    const showVirtualKeyboardIcon = (show) => {
+      if (document.getElementById('keyboardInputMaster')) {
+        return; // if keyboard is open, ignore
+      }
+      if (Sefaria.interfaceLang === 'english' && !hideHebrewKeyboard) {
+        const keyboardInitiator = document.querySelector(".keyboardInputInitiator");
+        if (keyboardInitiator) {
+          keyboardInitiator.style.display = show ? "inline" : "none";
+        }
+      }
+    };
+    const focusSearch = () => {
+      setSearchFocused(true);
+      showVirtualKeyboardIcon(true);
+    };
+
+    const blurSearch = (e) => {
+      onBlur(e);
+      const parent = document.getElementById('searchBox');
+      if (!parent.contains(e.relatedTarget) && !document.getElementById('keyboardInputMaster')) {
+        // debug: comment out the following line:
+        setSearchFocused(false);
+        showVirtualKeyboardIcon(false);
+      }
+    };
+
+    const inputClasses = classNames({
+      search: 1,
+      serif: 1,
+      keyboardInput: Sefaria.interfaceLang === "english",
+      hebrewSearch: Sefaria.interfaceLang === "hebrew"
+    });
+
+    const searchBoxClasses = classNames({ searchBox: 1, searchFocused });
+
+    return (
+      <div id="searchBox"
+           className={searchBoxClasses}>
+        <SearchButton onClick={handleSearchButtonClick} />
+        <input
+          className={inputClasses}
+          id="searchInput"
+          placeholder={Sefaria._("Search")}
+          onKeyDown={handleSearchKeyDown}
+          onFocus={focusSearch}
+          onBlur={blurSearch}
+          maxLength={75}
+          title={Sefaria._("Search for Texts or Keywords Here")}
+          {...otherDownShiftProps}
+        />
+      </div>
+    );
+  };
+const SuggestionsDispatcher = ({ suggestions, getItemProps, highlightedIndex,
+                                            submitSearch, redirectToObject}) => {
+
+    let groupedSuggestions = groupByType(suggestions);
+    let universalIndex = 0;
+
+    return (
+        <>
+            {groupedSuggestions.map((object, index) => {
+                const InitialIndexForGroup = universalIndex;
```

Comment:
make this variable lower case. In React, upper case variables are meant for components generally